app/src/main/java/com/dmonzonis/nookpendium/MainActivity.kt

```diff
@@ -0,0 +1,151 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        sharedPrefs = getSharedPreferences(getString(R.string.shared_prefs), Context.MODE_PRIVATE)!!
+
+        navGamesView.setNavigationItemSelectedListener(navListener())
+
+        val critterFragment = CritterFragment(sharedPrefs.getInt("selectedGame", R.string.game_acnh))
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.fragment_container, critterFragment)
+            .commit()
+
+        filterManager = FilterManager(critterFragment)
+        sortManager = SortManager(critterFragment)
+        setUpFilterManager()
+        setUpSortManager()
+
+        toolbar.inflateMenu(R.menu.toolbar_menu)
+        setSupportActionBar(toolbar)
+        supportActionBar!!.apply {
+            setDisplayShowTitleEnabled(false)
+            setDisplayHomeAsUpEnabled(true)
+            setHomeAsUpIndicator(R.drawable.ic_menu_white_24dp)
+        }
+    }
+
+    private fun navListener() : NavigationView.OnNavigationItemSelectedListener {
+        return NavigationView.OnNavigationItemSelectedListener {
+            var selectedFragment : Fragment? = null
+
+            when (it.itemId) {
+                R.id.miGameAcnh -> {
+                    selectedFragment = CritterFragment(R.string.game_acnh)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnh).apply()
+                }
+                R.id.miGameAcnl -> {
+                    selectedFragment = CritterFragment(R.string.game_acnl)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnl).apply()
+                }
+                R.id.miAbout -> selectedFragment = AboutFragment()
+            }
+            if (selectedFragment != null) {
+                supportFragmentManager.beginTransaction()
+                    .replace(R.id.fragment_container, selectedFragment).commit()
+                if (selectedFragment is CritterFragment) {
+                    filterManager.fragment = selectedFragment
+                    sortManager.fragment = selectedFragment
+                }
+            }
+            drawerLayout.closeDrawer(navGamesView)
+            true
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.toolbar_menu, menu)
+        // Set up search action
+        var searchMenuItem = menu?.findItem(R.id.miSearch)
+        val searchView = searchMenuItem?.actionView as SearchView
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                return false  // Do nothing, search is done on text change
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                filterManager.searchQuery = newText ?: ""
+                filterManager.fragment.recomputeFilters()
```

Comment:
since we only load one fragment at a time in the activity, we could keep track of the currently loaded fragment (make it a class var) and use that and in other parts where we may need the fragment in the future